src/lib/blockchain/Token.ts

```diff
@@ -77,4 +80,34 @@ export class Token {
 
     return externalEvents;
   }
+
+  public async approve(address: string, value: number) {
+    const gasPrice = await estimateGasPrice(this.web3);
+    const getWei = toWei(value.toString());
+    const approve = await this.contract.methods
+      .approve(address, getWei)
+      .send({ from: this.account, gasPrice });
+
+    return approve;
+  }
+
+  public async isTokenApprovedForFrogMint(attempts = 0): Promise<boolean> {
```

Comment:
Could we make this more reusable for other community projects and let the caller pass in the contract

I.e. `isTokenApprovedForContract(address: string, attempts = 0):`

Reply:
Good catch! I'll update this as well. ❤️